src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java

```diff
@@ -64,6 +64,10 @@ public User updateDepartment(Integer id, Integer departmentId) {
     public User updateRoles(Integer id, List<Integer> newRoles) {
         User user = userRepository.findById(id).orElseThrow(()-> new RuntimeException("User not found"));
         user.removeAllRoles();
+        if(newRoles == null || newRoles.isEmpty()) {
```

Comment:
There are quite a few conditionals in this method now.  Can you extract the FIND and APPLICANT role checking out to make it more readable?

Something like this:

`private void addRoleIfNotPresent(User user, RoleEnum roleName) {
        if (user.getRoles().stream().noneMatch(role -> role.getName().equals(roleName))) {
            Role role = roleRepository.findByName(roleName).orElseThrow(() -> new RoleNotFoundException(
                    "Update Roles failed: ".concat(roleName.name()).concat("role not found")));
            user.addRole(role);
        }
    }`

Reply:
Addressed in latest commit